python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -1223,8 +1245,467 @@ def asset_check_spec(self) -> AssetCheckSpec:
         return asset_checks_def.spec
 
 
-# actually forking the object type for assets is tricky for users in the cases of:
-#  * manually constructing ops to make AssetsDefinitions
-#  * having ops in a graph that form a graph backed asset
-# so we have a single type that users can call by their preferred name where appropriate
-AssetExecutionContext: TypeAlias = OpExecutionContext
+############################
+##### AssetExecutionContext
+############################
+
+# To preserve backwards compatibility, AssetExecutionContext is being written as a subclass of
+# OpExecutionContext until we can split it into its own class. All methods on OpExecutionContext
+# that will not be included in the eventual AssetExecutionContext will be marked with deprecation
+# warnings according to how the user should access that functionality in the future
+#
+# The following sets/maps are used to determine which methods need deprecation warnings, and how to
+# direct users to the correct method to use
+
+
+OP_EXECUTION_CONTEXT_ONLY_METHODS = set(
+    [
+        "describe_op",
+        "file_manager",
+        "has_assets_def",
+        "get_mapping_key",
+        "get_step_execution_context",  # TODO - used by internals
```

Comment:
TODO - put the methods used by internals in a different set and handle separately

Reply:
some options:
* leave them un-deprecated 
* wherever these methods are called in our internal code, do a 
``` python
def some_method(context):
     if isinstance(context, AssetExecutionContext):
           context = context._op_execution_context
```

Reply:
We should force ourselves to do the same process that any user will do on any deprecated method, which I suspect will be something like getting an inner context object.

Reply:
great - that's what i went with, and is mostly implemented in https://github.com/dagster-io/dagster/pull/16493 - i'll remove these comments